app/services/foreman_discovery/import_hooks/subnet_and_taxonomy.rb

```diff
@@ -4,35 +4,48 @@ module ImportHooks
     class SubnetAndTaxonomy < ImportHook
       def after_populate
         primary_ip = host.primary_interface.ip
+        primary_ip6 = host.primary_interface.ip6
 
         unless primary_ip
           logger.warn "Unable to assign subnet - reboot trigger may not be possible, primary interface is missing IP address"
           return
         end
 
         # set subnet
-        set_subnet(primary_ip)
+        set_subnets(primary_ip, primary_ip6)
         # set location and organization
         set_location
         set_organization
       end
 
       private
 
-      def set_subnet(ip)
-        host.primary_interface.subnet = suggested_subnet(ip)
+      def set_subnets(ip, ip6)
+        host.primary_interface.subnet = suggested_ipv4_subnet(ip)
+        subnet6 = suggested_ipv6_subnet(ip6) if ip6
+        logger.warn("Detected IPv6 subnet but it's not set for the host, unsupported") if subnet6
       end
 
-      def suggested_subnet(ip)
+      def suggested_ipv4_subnet(ip)
```

Comment:
Sorry what?

Reply:
What I meant was: `suggested_ipv4_subnet` and `suggested_ipv6_subnet` are basically the same methods, just a parameter is different. This might be a little overengineered, but I think it's a good idea to keep the hooks as clean as possible. And everything that increases the code quality in discovery is a good thing.

```ruby
# app/services/foreman_discovery/subnet_suggestion.rb
module ForemanDiscovery
  class SubnetSuggestion
    attr_accessor :ip, :kind

    def self.for(ip:, kind:)
      new(ip: ip, kind: kind).()
    end

    def initialize(ip:, kind:)
     @ip = ip
     @kind = kind
    end

    def call
      return unless ip
      subnet = Subnet.unscoped.subnet_for(ip)
      if subnet
        logger.info "Detected #{kind} subnet: #{subnet} with taxonomy #{subnet.organizations.collect(&:name)}/#{subnet.locations.collect(&:name)}"
      else
        logger.info "#{kind} subnet could not be detected for #{ip}"
      end
      subnet
    end
  end
end
```

So you can just call

```ruby
ForemanDiscovery::SubnetSuggestion.for(ip: ip, kind: 'IPv4')
```

And this can be tested with dedicated unit tests if this get's fancier in the future.